internal/formats/common/spdxhelpers/to_syft_model.go

```diff
@@ -0,0 +1,373 @@
+			Version:    version,
+			VersionID:  version,
+		}
+	}
+
+	return nil
+}
+
+func findLinuxReleaseByPURL(doc *spdx.Document2_2) *linux.Release {
+	for _, p := range doc.Packages {
+		purlValue := extractPURL(p)
+		if purlValue != "" {
+			purl, err := packageurl.FromString(purlValue)
+			if err != nil {
+				log.Warnf("unable to parse purl: %s", purlValue)
+			} else {
+				for _, qualifier := range purl.Qualifiers {
+					if qualifier.Key == "distro" {
+						parts := strings.Split(qualifier.Value, "-")
+						if len(parts) > 1 {
+							name := parts[0]
+							version := parts[1]
+							return &linux.Release{
+								PrettyName: name,
+								Name:       name,
+								ID:         name,
+								IDLike:     []string{name},
+								Version:    version,
+								VersionID:  version,
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func findSyftLinuxRelease(doc *spdx.Document2_2) *linux.Release {
+	r := findLinuxReleaseByDocument(doc)
+	if r != nil {
+		return r
+	}
+
+	r = findLinuxReleaseByPURL(doc)
+	if r != nil {
+		return r
+	}
+
+	return nil
+}
+
+func findSpdxRelationshipByType(doc *spdx.Document2_2, typ string) *spdx.Relationship2_2 {
+	for _, r := range doc.Relationships {
+		if typ == r.Relationship {
+			return r
+		}
+	}
+	return nil
+}
+
+func findSpdxPackageByID(doc *spdx.Document2_2, id spdx.ElementID) *spdx.Package2_2 {
+	for _, p := range doc.Packages {
+		if p.PackageSPDXIdentifier == id {
+			return p
+		}
+	}
+	return nil
+}
+
+func findSpdxReferenceByName(p *spdx.Package2_2, categories ...string) *spdx.PackageExternalReference2_2 {
+	for _, r := range p.PackageExternalReferences {
+		for _, category := range categories {
+			if r.Category == category {
+				return r
+			}
+		}
+	}
+	return nil
+}
+
+func collectSyftFiles(spdxIDMap map[string]interface{}, doc *spdx.Document2_2) {
+	for _, f := range doc.UnpackagedFiles {
+		spdxIDMap[string(f.FileSPDXIdentifier)] = toSyftLocation(f)
+	}
+}
+
+func toSyftRelationships(spdxIDMap map[string]interface{}, doc *spdx.Document2_2) []artifact.Relationship {
+	var out []artifact.Relationship
+	for _, r := range doc.Relationships {
+		// FIXME what to do with r.RefA.DocumentRefID and  r.RefA.SpecialID
+		if r.RefA.DocumentRefID != "" && requireAndTrimPrefix(r.RefA.DocumentRefID, "DocumentRef-") != string(doc.CreationInfo.SPDXIdentifier) {
+			log.Debugf("relationship to external document: %+v", r)
+			continue
+		}
+		a := spdxIDMap[string(r.RefA.ElementRefID)]
+		b := spdxIDMap[string(r.RefB.ElementRefID)]
+		from, fromOk := a.(*pkg.Package)
+		toPackage, toPackageOk := b.(*pkg.Package)
+		toLocation, toLocationOk := b.(*source.Location)
+		if !fromOk || !(toPackageOk || toLocationOk) {
+			log.Debugf("unable to find valid relationship mapping from SPDX 2.2 JSON, ignoring: (from: %+v) (to: %+v)", a, b)
+			continue
+		}
+		var to artifact.Identifiable
+		var typ artifact.RelationshipType
+		switch r.Relationship {
+		case "CONTAINS":
```

Comment:
nit: we could move the relationship constants from https://github.com/anchore/syft/blob/main/internal/formats/spdx22json/model/relationship.go to the SPDX helpers and use them instead of raw strings